### DISCUSSION TITLE
Is it possible to have optional fields?

### DISCUSSION THREAD

**Discussion by bogdankostic** [milvus-io/milvus#21457] [Q&A and General discussion]
I understand that we need to define a collection's schema before adding data and cannot modify it later (I think you are already working on this in https://github.com/milvus-io/milvus/issues/20405. Looking forward to it, I think this would be quite a useful feature!).

Regarding my question about optional fields, I would like to know if it is possible to have fields in the schema that are not required for each entity. For example, can we have a field that does not need to have a value for every entity, or can we set the value of a field for an entity to something like None or a placeholder value if the entity does not have a value for that field?

**Comment by xiaofan-luan** [comment]
I love the suggestion, and actually we are working on it.
We want to support list, set and probably some other dynamic containers.

Null value and dynamic schema change is another interesting topic, we should be able to finish the feature in 2023

**Reply by ndtreviv** [reply]
This would be great! Currently getting ParamExceptions because an optional field is sometimes null.
